### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by edubew**
### N + 1 problem:
- Solve the N+1 problem when fetching all posts and their comments for a user.
### **Before**      
n+1-before
### **After**     
n+1-after
 
### 2. Integration specs
- Use Capybara to write integration tests for each view in your project.

**1. User index page:**
- I can see the username of all other users.
- I can see the profile picture for each user.
-  I can see the number of posts each user has written.
- When I click on a user, I am redirected to that user's show page.

**2. user show page:**
- I can see the user's profile picture.
- I can see the user's username.
- I can see the number of posts the user has written
- I can see the user's bio.
- I can see the user's first 3 posts.
- I can see a button that lets me view all of a user's posts.
- When I click a user's post, it redirects me to that post's show page.
- When I click to see all posts, it redirects me to the user's post's index page.

**3. User post index page:**
- I can see the user's profile picture.
- I can see the user's username.
- I can see the number of posts the user has written.
- I can see a post's title.
- I can see some of the post's body.
- I can see the first comments on a post.
- I can see how many comments a post has.
- I can see how many likes a post has.
-  can see a section for pagination if there are more posts than fit on the view.
- When I click on a post, it redirects me to that post's show page.

**4. Post show page:**
- I can see the post's title.
- I can see who wrote the post.
- I can see how many comments it has.
- I can see how many likes it has.
- I can see the post body.
- I can see the username of each commentor.
- I can see the comment each commentor left.

### CODE REVIEW

File: spec/features/posts_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    end
+
+    it 'should show likes counter' do
+      expect(page).to have_content(@first_post.likes_counter)
+    end
+
+    it 'should navigate to post show page when clicked on post' do
+      click_on @first_post.title
+      expect(current_path).to eq(current_path)
+    end
```
Comment (jerryowusu) [inline spec/features/posts_index_spec.rb:49]: Nice work so far. However, you are missing some of the tests. For instance `Tests that asserts you can see a section for pagination if there are more posts than fit on the view.` is missing here. Kindly ensure that all tests cases have been covered. :+1:

File: spec/features/posts_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    @user1 = User.create(name: 'Sam', photo: 'https://icons.iconarchive.com/icons/iconsmind/outline/256/User-icon.png',
+                         bio: 'Teacher from Poland.', post_counter: 0)
+    visit users_path
+  end
+
+  scenario 'I can see the username, profile picture, and number of posts for each user' do
+    expect(page).to have_content(@user1.name)
+    expect(page.html).to include(@user1.photo)
+    expect(page).to have_content(@user1.post_counter)
+  end
```
Comment (jerryowusu) [inline spec/features/posts_show_spec.rb]: You are suppose to write at least 7 tests but I can see only 3 so far. Kindly ensure that all tests cases are covered as indicated in the project requirements. :pray: 

Screenshot from 2023-06-21 11-53-28

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi Team 

Good job so far!
Your project is looking fantastic so far :star_struck: 

However, there are few changes you need to make in order to move on to the next level :sunglasses:

soclosecat


## Highlights
Tests are passing :+1: 
Used capybara gem :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tugsem — APPROVED** [review]
## STATUS: APPROVED 👍🎯  
Hello @edubew  👋 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⭐ 
✔️ requested changes are implemented succesfully
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**